src/ProjectManagerClient.cs

```diff
@@ -333,9 +343,28 @@ public static ProjectManagerClient WithEnvironment(string env, HttpClientHandler
         /// <param name="customEndpoint">The custom endpoint to use for this client</param>
         /// <param name="clientHandler">Optional handler to set specific settings for the HTTP client</param>
         /// <returns>The API client to use</returns>
-        public static ProjectManagerClient WithCustomEnvironment(Uri customEndpoint, HttpClientHandler clientHandler = null)
+        public static ProjectManagerClient WithCustomEnvironment(Uri customEndpoint, HttpClientHandler clientHandler)
         {
-            return new ProjectManagerClient(customEndpoint, clientHandler);
+            return new ProjectManagerClient(customEndpoint, null, clientHandler);
+        }
+        
+        /// <summary>
+        /// Construct a client that uses a non-standard environment; this can be necessary when using proxy servers or
+        /// an API gateway.  Please be careful when using this mode.
+        /// You should prefer to use `WithEnvironment()` instead wherever possible.
+        /// </summary>
+        /// <param name="env">The named environment to use; should be "production"</param>
+        /// <param name="client">The HttpClient object to use</param>
+        /// <returns>The ProjectManager API client</returns>
+        public static ProjectManagerClient WithCustomHttpClient(string env, HttpClient client)
```

Comment:
Perfect, thanks!

What are your recommendations for test scenarios? To generate trial environment and execute against?

Reply:
We don't have official recommendations; I would defer to your testing team on how they would want to set up mocks.